pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -631,15 +631,20 @@ protected void startStatsUpdater(int statsUpdateInitialDelayInSecs, int statsUpd
     }
 
     protected void startDeduplicationSnapshotMonitor() {
+        // We do not know whether users will enable deduplication on namespace level/topic level or not, so keep this
+        // scheduled task runs.
         int interval = pulsar().getConfiguration().getBrokerDeduplicationSnapshotFrequencyInSeconds();
-        if (interval > 0 && pulsar().getConfiguration().isBrokerDeduplicationEnabled()) {
+        if (interval > 0) {
             this.deduplicationSnapshotMonitor = OrderedScheduler.newSchedulerBuilder()
                     .name("deduplication-snapshot-monitor")
                     .numThreads(1)
                     .build();
             deduplicationSnapshotMonitor.scheduleAtFixedRate(() -> forEachTopic(
```

Comment:
Is it make sense to let here be a safe run?(maybe it's another PR)

Reply:
It only will be executed once when the broker is starting, so it is fine

Reply:
No, I mean if `() -> forEachTopic()` method  throw the exception will stop the scheduler.

```
Submits a periodic action that becomes enabled first after the given initial delay, and subsequently with the given period; that is, executions will commence after initialDelay, then initialDelay + period, then initialDelay + 2 * period, and so on.
The sequence of task executions continues indefinitely until one of the following exceptional completions occur:
The task is explicitly cancelled via the returned future.
The executor terminates, also resulting in task cancellation.
An execution of the task throws an exception. In this case calling get on the returned future will throw ExecutionException, holding the exception as its cause.
Subsequent executions are suppressed. Subsequent calls to isDone() on the returned future will return true.
If any execution of this task takes longer than its period, then subsequent executions may start late, but will not concurrently execute.
```

Reply:
`OrderedScheduler` did the catch. See below:

<img width="830" alt="Screenshot 2024-04-11 at 15 03 32" src="https://github.com/apache/pulsar/assets/25195800/cac39b07-dea3-47b6-a2f0-36aa08249766">

<img width="898" alt="Screenshot 2024-04-11 at 15 04 31" src="https://github.com/apache/pulsar/assets/25195800/83dbba74-c2a9-4d54-b903-816e75853559">